src/main/java/software/amazon/awssdk/crt/s3/S3ClientOptions.java

```diff
@@ -100,6 +108,21 @@ public CredentialsProvider getCredentialsProvider() {
         return credentialsProvider;
     }
 
+    /**
+     * The configuration related to signing used by S3 client. It will override the credentials provider if provided.
+     * `AwsSigningConfig.getDefaultS3SigningConfig(region, credentialsProvider);` can be used as helper to create the default configuration to be used for S3.
+     * @param signingConfig configuration related to signing via an AWS signing process.
+     * @return this
+     */
```

Comment:
debatable: I don’t like the idea of having multiple ways to do things and priority is defined by the order in which we added them. What happens if we need to add a third way in future? We should just error out if multiple things are set.

Reply:
I marked the credentials provider as deprecated to keep it consist with our C level, and as we have the helper function to set up the default signing config, is not that bad